### PR TITLE
 Add telemetry for spring maven plugin

### PR DESCRIPTION
**Pull request by Flanker32**
Add telemetry for spring maven plugin
Update AI to 2.5.0-Beta.2 which support Java 11

### CODE REVIEW

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/AbstractSpringMojo.java [hunk 1]
```diff
@@ -57,9 +76,72 @@
...
+            doExecute();
+            handleSuccess();
+        } catch (Exception e) {
+            handleException(e);
+        } finally {
+            // When maven goal executes too quick, The HTTPClient of AI SDK may not fully initialized and will step
+            // into endless loop when close, we need to call it in main thread.
+            // Refer here for detail codes: https://github.com/Microsoft/ApplicationInsights-Java/blob/master/core/src
+            // /main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender43.java#L103
+            try {
```
Comment (andxu) [inline azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/AbstractSpringMojo.java]: MOVE the strange code to AppInsightHelper

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/AbstractSpringMojo.java [hunk 2]
```diff
@@ -57,9 +76,72 @@
...
+    }
+
+    protected void initTelemetry() {
+        telemetries = new HashMap<>();
+        telemetries.put(TELEMETRY_KEY_PLUGIN_NAME, plugin.getArtifactId());
+        telemetries.put(TELEMETRY_KEY_PLUGIN_VERSION, plugin.getVersion());
+        telemetries.put(TELEMETRY_KEY_WITHIN_PARENT_POM, String.valueOf(project.getPackaging().equalsIgnoreCase("pom")));
+    }
+
+    protected void updateTelemetry(SpringConfiguration configuration) {
```
Comment (andxu) [inline azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/AbstractSpringMojo.java]: traceConfiguration?

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java [hunk 3]
```diff
@@ -40,4 +40,15 @@ protected void doExecute() throws MojoExecutionException, MojoFailureException {
         // Update deployment, show url
         getLog().info(app.properties().url());
     }
+
+    @Override
+    protected void updateTelemetry(SpringConfiguration springConfiguration) {
```
Comment (andxu) [inline azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java]: don't Override updateTelemetry, but a new method traceAuth and traceUsage

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java [hunk 4]
```diff
@@ -40,4 +40,15 @@ protected void doExecute() throws MojoExecutionException, MojoFailureException {
...
+        super.updateTelemetry(springConfiguration);
+        // Todo update deploy mojo telemetries with real value
+        telemetries.put("authMethod", "Pom Configuration");
+        telemetries.put("isServicePrincipal", "false");
```
Comment (andxu) [inline azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java]: wrap it in a TelemetryConstant

Comment (Flanker32) [reply azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java]: Done

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/AbstractSpringMojo.java (hunk 2 above)
Comment (Flanker32) [reply azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/AbstractSpringMojo.java]: Done.

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/AbstractSpringMojo.java (hunk 1 above)
Comment (Flanker32) [reply azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/AbstractSpringMojo.java]: Move to the `close()` method of AppInsightHelper

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java (hunk 3 above)
Comment (Flanker32) [reply azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java]: rename to `traceTelemetry`

**Pull request merged**